src/read-coverage.js

```diff
@@ -0,0 +1,47 @@
+import genVar from './gen-var';
+import { parse } from 'babylon';
+import traverse from 'babel-traverse';
+import * as t from 'babel-types';
+
+export default function readInitialCoverage (code, filename) {
+    if (typeof code !== 'string') {
+        throw new Error('Code must be a string');
+    }
+
+    // Parse as leniently as possible
+    const ast = parse(code, {
+        allowImportExportEverywhere: true,
+        allowReturnOutsideFunction: true,
+        allowSuperOutsideMethod: true,
+        sourceType: "script", // I think ?
+        plugins: ["*"]
+    });
+
+    const varName = genVar(filename);
+    let covPath;
+    traverse(ast, {
+        VariableDeclarator: function (path) {
```

Comment:
😀 

Just to be more precise here:

We parse the entire file, then stop _traversing_ the AST the moment we find a variable declaration whose name is equal to `genVar`. Then we use `babel-traverse`'s robust understanding of scopes and ability to statically evaluate nodes (much the opposite of `valueToNode` used [here](https://github.com/istanbuljs/istanbul-lib-instrument/blob/0225e8cbe3777193d874076c7dc75f1fd567ccc3/src/visitor.js#L469)) to extract the value of `coverageData` from within that IIFE.

The change to `babel-traverse` merely adds support for evaluating object literals, which was a long-standing `TODO` in Babel and of course is essential here.